src/services/permissions/utils/PermissionCheckUtils.ts

```diff
@@ -0,0 +1,33 @@
+import PrisonerPermissionsContext from '../../../types/internal/permissions/PrisonerPermissionsContext'
+import { PrisonerPermission } from '../../../types/public/permissions/prisoner/PrisonerPermissions'
+import { getPermissionStatus, PrisonerPermissionConditions } from '../PrisonerPermissionConditions'
+import { PermissionCheckStatus } from '../../../types/internal/permissions/PermissionCheckStatus'
+import { baseCheckConditions } from '../checks/baseCheck/status/BaseCheckStatus'
+import { logDeniedPermissionCheck } from './PermissionUtils'
+
+export const matchBaseCheckAnd =
+  (permissionRules: Partial<PrisonerPermissionConditions>) =>
+  (permission: PrisonerPermission, context: PrisonerPermissionsContext) => {
+    const { user, prisoner, baseCheckStatus } = context
+    const baseCheckPassed = baseCheckStatus === PermissionCheckStatus.OK
+
+    const permissionStatus = getPermissionStatus(user, prisoner, {
+      ...baseCheckConditions,
+      ...permissionRules,
+    })
+
+    const permissionCheckPassed = baseCheckPassed && permissionStatus === PermissionCheckStatus.OK
+
+    if (!permissionCheckPassed) logDeniedPermissionCheck(permission, context, permissionStatus)
+
+    return permissionCheckPassed
+  }
+
+export const checkWith =
```

Comment:
This is a bit hefty on the typescript generics but it makes the construction of the permissions object neater and retains type safety, making sure all permissions checks are present